src/test/java/teammates/test/cases/BaseComponentTestCase.java

```diff
@@ -65,164 +47,82 @@ protected static boolean doesFileExistInGcs(BlobKey fileKey) {
-                expectedCopy.feedbackQuestionId, expectedCopy.giver, expectedCopy.recipient);
-        if (wildcardId) {
-            expectedCopy.setId(actual.getId());
-        }
-        
-        assertEquals(JsonUtils.toJson(expectedCopy), JsonUtils.toJson(actual));
+    @Override
+    protected InstructorAttributes getInstructor(InstructorAttributes instructor) {
+        return instructor.googleId == null
+                ? backDoorLogic.getInstructorForEmail(instructor.courseId, instructor.email)
+                : backDoorLogic.getInstructorForGoogleId(instructor.courseId, instructor.googleId);
     }
     
-    protected static void verifyAbsentInDatastore(FeedbackSessionAttributes fs) {
-        assertNull(fsDb.getFeedbackSession(fs.getCourseId(), fs.getFeedbackSessionName()));
+    @Override
+    protected StudentAttributes getStudent(StudentAttributes student) {
+        return backDoorLogic.getStudentForEmail(student.course, student.email);
     }
     
-    protected static void verifyPresentInDatastore(FeedbackSessionAttributes expected) {
-        FeedbackSessionAttributes expectedCopy = expected.getCopy();
-        FeedbackSessionAttributes actual =
-                fsDb.getFeedbackSession(expectedCopy.getCourseId(), expected.getFeedbackSessionName());
-        expectedCopy.setRespondingInstructorList(actual.getRespondingInstructorList());
-        expectedCopy.setRespondingStudentList(actual.getRespondingStudentList());
-        assertEquals(JsonUtils.toJson(expectedCopy), JsonUtils.toJson(actual));
-    }
-
-    protected static void verifyAbsentInDatastore(InstructorAttributes instructor) {
-        assertNull(instructorsDb.getInstructorForGoogleId(instructor.courseId, instructor.googleId));
-    }
-
-    protected static void verifyPresentInDatastore(InstructorAttributes expected) {
-        InstructorAttributes expectedCopy = expected.getCopy();
-        InstructorAttributes actual = expected.googleId == null
-                                    ? instructorsDb.getInstructorForEmail(expectedCopy.courseId, expected.email)
-                                    : instructorsDb.getInstructorForGoogleId(expectedCopy.courseId, expected.googleId);
-        equalizeIrrelevantData(expectedCopy, actual);
-
-        assertTrue(expectedCopy.isEqualToAnotherInstructor(actual));
+    protected void removeAndRestoreTypicalDataBundle() {
+        DataBundle dataBundle = getTypicalDataBundle();
+        removeAndRestoreDataBundle(dataBundle);
     }
     
-    private static void equalizeIrrelevantData(InstructorAttributes expectedInstructor,
-                                               InstructorAttributes actualInstructor) {
-        
-        // pretend keys match because the key is generated only before storing into database
-        if (actualInstructor.key != null) {
-            expectedInstructor.key = actualInstructor.key;
+    @Override
+    protected String doRemoveAndRestoreDataBundle(DataBundle dataBundle) {
```

Comment:
If we're using the word ```do``` to describe the method, perhaps we should do the same for the method ```removeAndRestoreTypicalDataBundle```? What do you think?